src/pages/BooksPage.js

```diff
@@ -28,10 +28,6 @@ const BooksPage = props => {
   const initialCoords = {lat: 0, lng: 0};
   const [totalPages, setTotalPages] = useState(1);
 
-  useEffect(() => {
```

Comment:
I don't know whether this one is unnecessary -- I hope that you have checked it thoroughly.

Reply:
surely no. We do the initial fetch in that   if( conditions )

Reply:
You shouldn't assume me didn't check and doubt my change out of no evidence and inspection upon current code.

I did this change after several check and experiments.